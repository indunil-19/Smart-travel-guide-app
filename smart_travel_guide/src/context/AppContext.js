import React, {createContext, useReducer} from 'react';
import { initialState, reducer as AppReducer } from '../reducer/AppReducer';

export const AppContext =createContext()

export const AppProvider=({children})=>{
    const [state, dispatch]= useReducer(AppReducer, initialState)

    return(
        <AppContext.Provider  value={{state,dispatch}}>
                {children}
        </AppContext.Provider>
    )
}