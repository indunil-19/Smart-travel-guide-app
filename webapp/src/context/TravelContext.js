import React, {createContext, useReducer} from 'react';
import { initialState, Travelreducer } from '../reducer/TravelReducer';

const TravelContext =createContext()

const TravelProvider=({children})=>{
    const [state, dispatch]= useReducer(Travelreducer, initialState)

    return(
        <TravelContext.Provider  value={{state,dispatch}}>
                {children}
        </TravelContext.Provider>
    )
}

export  {TravelContext,  TravelProvider}
 