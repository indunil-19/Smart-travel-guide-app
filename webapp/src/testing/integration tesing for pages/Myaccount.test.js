import React from "react"
import {render,screen, cleanup,waitFor,fireEvent} from "@testing-library/react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import { MemoryRouter,Router } from "react-router-dom";
import {TravelProvider} from "../../context/TravelContext"
import "@testing-library/jest-dom/extend-expect";
import { rest } from "msw";
import { setupServer } from "msw/node";
import 'whatwg-fetch'
import MyAccount from "../../screens/TravelPlanApp/myAccount";
import { createMemoryHistory } from 'history'

describe('my account page', () => {
    


    const server=setupServer(
        rest.post("/user/updateUser",(req,res,ctx)=>{
            return res(
                ctx.json({
                    _id:req.body._id,
                    firstname:req.body.firstname,
                    lastname:req.body.lastname,
                    dob:req.body.dob,
                    country:req.body.country,
                    religion:req.body.religion,
                    email:req.body.email, 
                    pic:req.body.pic

                    })
            )
        }),
    )
 
    beforeEach(()=>{      
    })
    beforeAll(() => server.listen());
    afterEach(() => server.resetHandlers());
    afterAll(() => server.close());
    afterEach(cleanup)

    it("should match snapshot",  () => {
        const tree = renderer.create(<MemoryRouter><TravelProvider><MyAccount/></TravelProvider></MemoryRouter>);
        expect(tree).toMatchSnapshot();
    });

    it("shoud render without crashing", ()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><MyAccount  /></TravelProvider></MemoryRouter>,div)
    })

    it("should enable edit if edit button click",()=>{
        const div = document.createElement("div");
        ReactDOM.render(<MemoryRouter><TravelProvider><MyAccount  /></TravelProvider></MemoryRouter>,div)
        waitFor(()=>{
            fireEvent.click(screen.getByTestId("edit"))
            waitFor(()=>{
                expect(screen.getByTestId("firstname")).toBeEnabled()
                expect(screen.getByTestId("lastname")).toBeEnabled()
                expect(screen.getByTestId("dob")).toBeEnabled()

                waitFor(()=>{
                    fireEvent.change(screen.getByTestId("firstname"),{target:{value:"indunil"}})
                    fireEvent.change(screen.getByTestId("lastname"),{target:{value:"udayngana"}})
                    fireEvent.change(screen.getByTestId("dob"),{target:{value:"1998/10/10"}})
                    fireEvent.change(screen.getByTestId("lastname"),{target:{value:"udayngana"}})
                    fireEvent.select(screen.getByTestId("religion"),{target:{value:"Buddhism"}})
                    fireEvent.select(screen.getByTestId("email"),{target:{value:"udayangana98@gmail.com"}})
                    fireEvent.click(screen.getByTestId("save"))
                })
            })
            
        })

    })


})
