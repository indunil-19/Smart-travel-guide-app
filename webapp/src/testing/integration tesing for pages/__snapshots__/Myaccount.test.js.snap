// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my account page should match snapshot 1`] = `
Array [
  <div
    className="css-1lgg032"
  >
    <input
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div
      className="chakra-stack css-84zodg"
    >
      <img
        className="chakra-image__placeholder css-1oidk5m"
      />
    </div>
    <div
      className="chakra-input__group css-1or4oaq"
    >
      <div
        className="chakra-input__left-addon css-1lhl5nw"
      >
        First Name
      </div>
      <input
        className="chakra-input css-fu3wfu"
        data-testid="firstname"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="First Name"
        type="text"
      />
    </div>
    <div
      className="chakra-input__group css-1or4oaq"
    >
      <div
        className="chakra-input__left-addon css-1lhl5nw"
      >
        Last Name
      </div>
      <input
        className="chakra-input css-fu3wfu"
        data-testid="lastname"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Last Name"
        type="text"
      />
    </div>
    <div
      className="chakra-input__group css-1or4oaq"
    >
      <div
        className="chakra-input__left-addon css-1lhl5nw"
      >
        Birth Day
      </div>
      <input
        className="chakra-input css-fu3wfu"
        data-testid="dob"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Birth Day"
        type="date"
      />
    </div>
    <div
      className="chakra-input__group css-1or4oaq"
    >
      <div
        className="chakra-input__left-addon css-1lhl5nw"
      >
        Religion
      </div>
      <div
        className="chakra-select__wrapper css-1uadmkv"
      >
        <select
          className="chakra-select css-1ik61og"
          data-testid="religion"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
        >
          <option
            value=""
          >
            religion
          </option>
          <option
            value="Buddhism"
          >
            Buddhism
          </option>
          <option
            value="Hindu"
          >
            Hindu
          </option>
          <option
            value="Islam"
          >
            Islam
          </option>
          <option
            value="Catholic"
          >
            Catholic
          </option>
          <option
            value=""
          >
            None
          </option>
        </select>
        <div
          className="chakra-select__icon-wrapper css-s9xp1k"
          data-disabled={true}
        >
          <svg
            aria-hidden={true}
            className="chakra-select__icon"
            focusable={false}
            role="presentation"
            style={
              Object {
                "color": "currentColor",
                "height": "1em",
                "width": "1em",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="chakra-input__group css-1or4oaq"
    >
      <div
        className="chakra-input__left-addon css-1lhl5nw"
      >
        Country
      </div>
      <input
        className="chakra-input css-fu3wfu"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Country"
        type="text"
      />
    </div>
    <div
      className="chakra-input__group css-1or4oaq"
    >
      <div
        className="chakra-input__left-addon css-1lhl5nw"
      >
        Email
      </div>
      <input
        className="chakra-input css-fu3wfu"
        data-testid="email"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Country"
        type="email"
      />
    </div>
    <div
      className="css-1d2ayld"
    >
      <button
        className="chakra-button css-5c2176"
        data-testid="edit"
        onClick={[Function]}
        type="button"
      >
        <span
          className="chakra-button__icon css-1wh2kri"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              stroke="currentColor"
              strokeLinecap="round"
              strokeWidth="2"
            >
              <path
                d="M11 4H4a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
              />
              <path
                d="M18.5 2.5a2.121 2.121 0 0 1 3 3L12 15l-4 1 1-4 9.5-9.5z"
              />
            </g>
          </svg>
        </span>
        Edit
      </button>
    </div>
  </div>,
  <div
    className="css-1lgg032"
  >
    <h3
      className="chakra-heading css-1o3848f"
    >
      Do you need to change your password???
    </h3>
    <a
      href="/travelPlan/changePassword"
      onClick={[Function]}
    >
      <button
        className="chakra-button css-n93402"
        type="button"
      >
        Click Here
      </button>
    </a>
  </div>,
]
`;
